ILC/CLIC difermion theory files

arossia94 opened the pull request:
I've added the theory prediction files for light difermion production at ILC/CLIC. They include the information of dynamic scales. Each "bin" is a different polarization configuration.

jacoterh commented:
Hi @moatms @arossia94 , many thanks for adding these projections! It's great to have them. I merged main into here and ran the new pre-commit. Currently the tests don't pass as the jsons miss the required "scales" key, which is needed for RG evolution. So if these jsons could just be extended with `scales: [scale_1, scale_2, ..., scale_ndat]` that would be great.

jacoterh commented:
@moatms RE the failing test, it can be resolved if you run locally `pre-commit run --all-files` and git stage and commit again the files that needed reformatting

LucaMantani commented:
@moatms You are somehow committing but github is not recognising your account.
can you run 
`git config user.email`
in the repo? what does it return?

moatms commented:
@LucaMantani Yes so I checked and basically my account email is one version of my manchester email while the `git config` email was another version... It should be fixed now with the same email for both

jacoterh commented:
Just for the record, we agreed to present tables split by polarisation instead of energy. The idea is to let smefit filter out measurements based on the scales entry.

arossia94 commented:
> Just for the record, we agreed to present tables split by polarisation instead of energy. The idea is to let smefit filter out measurements based on the scales entry.

So the different energies would be like "bins" with different scales, do I understand well? If so, I'll reprint the ILC/CLIC difermion predictions in that format in the coming days.

jacoterh commented:
> > Just for the record, we agreed to present tables split by polarisation instead of energy. The idea is to let smefit filter out measurements based on the scales entry.
> 
> So the different energies would be like "bins" with different scales, do I understand well? If so, I'll reprint the ILC/CLIC difermion predictions in that format in the coming days.

Yes, exactly :)

arossia94 reviewed (commented):
@moatms , the ILC at 1TeV has a different polarization setup than at lower energies: it's +-80+-20, instead of +-80+-30 (see Table 11 in Snowmass). I noticed you grouped all together, it should be separated (see how I did it for difermion). And most importantly, check that the computation at 1 TeV was done with the right +-80+-20 polarizations.

moatms commented:
@arossia94  I put all the runs together to have less files, but the 1TeV runs were done indeed with +-80+-20 so the results are fine

arossia94 commented:
The following changes have been made:

- Finished and implemented the ee>tt OOs for ILC/CLIC
- Homogenized the implementation of the OOs for tt/WW and all the colliders. Explanation to follow in another comment.
- Rebased the branch to current main in order to incorporate NLO-EW corrections in ZH and other changes such as the NLO_QCD and NLO_EW order keys. This is required to run up-to-date fits with ILC/CLIC on top of HL-LHC.

arossia94 commented:
I added the possibility of using the $e^{+} e^{-}  \to t t$ Optimal Observables with the full basis that includes 2-lepton 2-quark operators.

These are implemented with the classes `Optimaltt{coll}{energy}full`.
The colliders and energies availables for these OOs (not every combination) are:
-`coll= {FCC, ILC, CLIC, CEPC}`
-`energy={161, 240, 250, 350, 365, 380, 500, 1000, 1500, 3000}`
Notice the ending `full` that distinguishes them from their counterparts with the reduced 4-operator basis.

The order and notation of the extended basis is:
`["OpQM", "Opt", "OtW", "OtZ", "Ol1QM", "OeQ", "Ol1t", "Oet"]`

jacoterh commented:
Is this ready for review? I'm not sure as it's still a draft.
